src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        HashMap<String, Integer> data = new HashMap<>();
```

Comment:
The expected solution for this exercise is simpler than you might expect, based on your comment.

First things first: Please read the file as-is into string, no pre-processing is needed here.

I like the way that you split the process into separate steps. You can make three private methods:
```java
String readFile(fromFileName);
String createReport(data);
void writeFile(report, toFileName);
```